### PR TITLE
Backport 1.13 refresh rate feature / fix.

### PR DESCRIPTION
**Pull request by Poltergeist**
This is a Backport of #4140 

## Testing

Go to ui Settings, 
Change the setting to something different then 2 seconds  ( bigger is better )
Reload the page.
Open network tab.
Open services page.
Check in the network tab that the endpoints are only requested at about the changed refresh rate setting.

You can also check this on various pages in the app.

## Trade-offs

~- You have to hard reload the page to use the new refresh rate on streams~

## Dependencies

n/a

## Screenshots

n/a

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
It looks like tests are actually failing.

File: src/js/stores/UserSettingsStore.ts [hunk 1]
```diff
@@ -2,19 +2,23 @@ import EventEmitter from "events";
 
 import LocalStorageUtil from "../utils/LocalStorageUtil";
 
-const LOCAL_STORAGE_KEY = "dcosUserSettings";
+const LOCAL_STORAGE_KEY: string = "dcosUserSettings";
+const REFRESH_RATE_SETTING: string = "RefreshRate";
```
Comment (GeorgiSTodorov) [inline src/js/stores/UserSettingsStore.ts]: I think these two don't need types.

File: src/js/stores/UserSettingsStore.ts [hunk 2]
```diff
@@ -2,19 +2,23 @@ import EventEmitter from "events";
...
+    } catch (e) {
+      return null;
+    }
   }
+  return null;
 }
 
 class UserSettingsStore extends EventEmitter {
-  getKey(key) {
+  getKey(key: string): unknown {
```
Comment (GeorgiSTodorov) [inline src/js/stores/UserSettingsStore.ts]: Is the return type really `unknown`?

Comment (Poltergeist) [reply src/js/stores/UserSettingsStore.ts]: Yes since it is a catchall function it can basically return anything.

**Comment by GeorgiSTodorov** [comment]
It works, I will approve once the tests are fixed (the easiest way is by changing the exports).

**Pull request merged**